client/reporter.go

```diff
@@ -17,4 +17,6 @@ type Reporter interface {
 	PostInstanceFail(ctx context.Context, id string, body io.Reader) error
 	PostInstancePassword(ctx context.Context, id, pass string) error
 	UpdateInstance(ctx context.Context, id string, body io.Reader) error
+
+	Post(ctx context.Context, ref, mime string, body io.Reader, v interface{}) error
```

Comment:
Does this need to be an exposed method? Can this be an internal implementation on the `cacher.HardwareFinder`? It looks like this is only called in the `cacher` package and not by external callers

Reply:
Hmm yeah thats a good point, I had this here before I figured out how to test it and thought I might need noop too. Will fix.

Reply:
hmm well actually dropping `Post` from `Reporter` sucks for other reasons. I've got 3 choices and they all have their warts.

1. Add Post to Reporter
2. Make cacher.NewHardwareFinder take a Poster and have `getFinders` type assert and bail if the Reporter isn't also a Poster
3. Leave everything as-is and make cacher.NewHardwareFinder do a runtime check and return an error.

1. is mostly unnecessary
2. we can't use Reporters everywhere
3. is going around the compiler/type system where it could help.

I was originally going to go with 3, but now that I think about it find 2 the better option. This is all terribly ugly/annoying anyway.

Reply:
I've reworked so Reporter no longer has a `Post` method. How's this look to you?